src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -30,4 +33,8 @@ target_link_libraries(System.Net.Security.Native
     ${LIBGSS}
 )
 
+include_directories(SYSTEM
+    ${LIBGSS_INCLUDE_DIR}
```

Comment:
> Doesn't this append the ${LIBGSS_INCLUDE_DIR} to the current set of included-directories?

Yes. This is the goal of this patch.

> Am I right in understanding that the heimdal version of gssapi is not present in the standard include path?

Heimdal gssapi is available freely in /usr/include

```
$ ls /usr/include/gssapi<TAB>
gssapi.h  gssapi/
$ ls /usr/include/gssapi/
gssapi.h         gssapi_krb5.h    gssapi_ntlm.h    gssapi_oid.h     gssapi_spnego.h
```

pkgsrc ships with mit-krb5:

```
$ pkg_info -L mit-krb5|grep include
/usr/pkg/include/com_err.h
/usr/pkg/include/gssapi.h
/usr/pkg/include/gssapi/gssapi.h
/usr/pkg/include/gssapi/gssapi_ext.h
/usr/pkg/include/gssapi/gssapi_generic.h
/usr/pkg/include/gssapi/gssapi_krb5.h
/usr/pkg/include/gssapi/mechglue.h
/usr/pkg/include/gssrpc/auth.h
/usr/pkg/include/gssrpc/auth_gss.h
/usr/pkg/include/gssrpc/auth_gssapi.h
/usr/pkg/include/gssrpc/auth_unix.h
/usr/pkg/include/gssrpc/clnt.h
/usr/pkg/include/gssrpc/netdb.h
/usr/pkg/include/gssrpc/pmap_clnt.h
/usr/pkg/include/gssrpc/pmap_prot.h
/usr/pkg/include/gssrpc/pmap_rmt.h
/usr/pkg/include/gssrpc/rename.h
/usr/pkg/include/gssrpc/rpc.h
/usr/pkg/include/gssrpc/rpc_msg.h
/usr/pkg/include/gssrpc/svc.h
/usr/pkg/include/gssrpc/svc_auth.h
/usr/pkg/include/gssrpc/types.h
/usr/pkg/include/gssrpc/xdr.h
/usr/pkg/include/kadm5/admin.h
/usr/pkg/include/kadm5/chpass_util_strings.h
/usr/pkg/include/kadm5/kadm_err.h
/usr/pkg/include/kdb.h
/usr/pkg/include/krad.h
/usr/pkg/include/krb5.h
/usr/pkg/include/krb5/ccselect_plugin.h
/usr/pkg/include/krb5/clpreauth_plugin.h
/usr/pkg/include/krb5/hostrealm_plugin.h
/usr/pkg/include/krb5/kadm5_hook_plugin.h
/usr/pkg/include/krb5/kdcpreauth_plugin.h
/usr/pkg/include/krb5/krb5.h
/usr/pkg/include/krb5/localauth_plugin.h
/usr/pkg/include/krb5/locate_plugin.h
/usr/pkg/include/krb5/plugin.h
/usr/pkg/include/krb5/preauth_plugin.h
/usr/pkg/include/krb5/pwqual_plugin.h
/usr/pkg/include/profile.h
/usr/pkg/include/verto-module.h
/usr/pkg/include/verto.h
```


> I am worried about the following : pal_gssapi.cpp includes gssapi/gssapi_krb5.h. This is safe because this is a file that is unique to mit libraries. But gssapi_krb5.h in turn includes gssapi/gssapi.h.

Yes, there might be conflicts, like mit-krb5 headers including the Heimdal ones.

> With the above `include_directories` isn't it possible that the file included is from heimdal version but is linked to mit-krb5 version?

No, because I'm searching for mit-krb5 specific header.